core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooReaderWriter.java

```diff
@@ -138,6 +128,24 @@ public void recursiveCopyPersistentOverwrite(String source, String destination)
     }, e -> e.code() == Code.BADVERSION); // always retry if bad version
   }
 
+  public byte[] createPublicOrMutate(String zPath, byte[] createValue, Mutator mutator)
```

Comment:
Should describe what this method is doing.  While this is an improvement over the previous method, it is not clear that it has anything to do with ACL's.  Perhaps mention it on one of the other methods?  I am not sure but we should describe somewhere what we are doing with the permissions on the node.

Reply:
The original method that I split didn't have a javadoc, presumably because it's only used internally and its function is obvious from the method signature. However, I can add a basic javadoc to both the two new methods to help clarify the purpose, and to make it more obvious what the permissions will be on the created method.

Reply:
I was thinking whatever methods end up being the top level or entry methods to these classes would be good.  No need to javadoc all the methods, especially private ones.

Reply:
I was only going to add the javadoc to the two I split this original one into. I'm not going to be an overachiever :smiley_cat:

Reply:
I ended up updating all the javadocs in this class... :smiley_cat: